hips/hip-00XX.md

```diff
@@ -0,0 +1,213 @@
+
+In a future version of Helm 4 CLI, it might be desirable to omit a warning if a user is not using SSA (in order to prompt the user to enabling SSA for further releases).
+
+#### Custom resources
+
+Helm will utilize SSA for custom resources, which it previously [didn't merge](https://github.com/helm/helm/blob/1b260d0a796882a1e90f0fa3c832659dbe2e675c/pkg/kube/converter.go#L54-L69).
+As SSA allows custom resources to be correctly merged based on metadata within the CRD.
+Or the API server will use the default approach (<https://kubernetes.io/docs/reference/using-api/server-side-apply/#custom-resources>).
+
+#### Conflicts and forcing
+
+It is possible that when Helm upgrades (or rollbacks) a chart, there will be a field conflict with another field manager.
+In this case, Helm will report the error for the conflict field(s) to the user.
+
+Helm will also add a new flag `--force-conflicts=false|true` to the install, upgrade and rollback commands (modeled after the life `kubectl apply` flag)
+And corresponding API objects field.
+When `true` (default `false`), Helm will enabling override the field conflicts, setting field(s) to the charts values.
+And also make Helm the field manager for the conflicting field(s).
+The `--force-conflicts` flag documentation should explicity mention the downsides of conflict forcing ie. the chart update is overriding object values that another process is expecting to own/manage.
+
+For more details, see the _Overwrite value, become sole manager_ of <https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts>
+
+(Of note, Helm upgrade already includes the `--force` flag, which tells Helm to replace objects during upgrade.
+However, given this different semantics of `--force`, a new flag is required)
+
+An additional special case is rollbacks on failed upgrade (`--atomic` flag). It is possible for Helm to attempt to rollback to a version which now has a conflict.
+For which the rollback will fail.
+Since rollbacks are "best effort" and can fail for other reasons as well, it seems best to treat this new rollback failure-case similarly (report to the user via logging and error).
+If the user specified `--force-conflicts` to the upgrade, the corresponding rollback should similarly force conflicts (this is the behavior of `--force`).
+If the rollback is to a release version which previously didn't have SSA enabled, the rollback install should also disable SSA.
+
+#### Special
+
+- `--dry-run=client` won't work with SSA enabled
+- If the user attempts to use SSA with a cluster which does not support it (unlikely: SSA [went GA](https://kubernetes.io/blog/2021/08/06/server-side-apply-ga/) in Kubernetes v1.22), Helm should error
+- Client-side API object validation isn't applicable when SSA enabled (`helm install|upgrade --disable-openapi-validation`)
```

Comment:
Could we still do this using the SSA with a dry-run?

Reply:
It just means dry run won't be able to accurately show the results of server side apply without connecting to the server. This is already often true with the current merge strategy, but it's especially true of server side apply strategy. We could do this though with `--dry-run=server`. 

From the docs:
> --dry-run string[="client"]                  simulate an install. If --dry-run is set with no option being specified or as '--dry-run=client', it will not attempt cluster connections. Setting '--dry-run=server' allows attempting cluster connections.